cmd/lambda/accounts/update.go

```diff
@@ -2,116 +2,53 @@ package main
-	Metadata            *map[string]interface{} `json:"metadata"`
-}
-
+// UpdateAccountByID updates an accounts information based on ID
 func UpdateAccountByID(w http.ResponseWriter, r *http.Request) {
 	accountID := mux.Vars(r)["accountId"]
 
 	// Deserialize the request JSON as an request object
-	var request *updateAccountRequest
+	d := model.Account{}
 	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&request)
+	err := decoder.Decode(&d)
 	if err != nil {
-		response.WriteAPIErrorResponse(w, http.StatusBadRequest, "ClientError", "invalid request parameters")
+		api.WriteAPIErrorResponse(w,
+			errors.NewBadRequest("invalid request parameters"))
 		return
 	}
-	request.ID = &accountID
 
-	// If the request includes a new adminRoleArn,
-	// validate that we can assume the ARN
-	if request.AdminRoleArn != nil {
-		_, err = TokenSvc.AssumeRole(&sts.AssumeRoleInput{
-			RoleArn:         request.AdminRoleArn,
-			RoleSessionName: aws.String("MasterAssumeRoleVerification"),
-		})
-
-		if err != nil {
-			response.WriteRequestValidationError(
-				w,
-				fmt.Sprintf("Unable to update account %s: "+
-					"admin role is not assumable by the master account",
-					accountID),
-			)
-			return
-		}
+	var dataSvc dataiface.AccountData
+	if err := Services.Config.GetService(&dataSvc); err != nil {
+		api.WriteAPIErrorResponse(w, err)
+		return
 	}
 
-	// Figure out which DB fields we want to update
-	// Note that not all fields are updatable
-	// eg, we don't want clients to set their own timestamps,
-	// 	or modify fields, like AccountStatus, which are  used for
-	// 	internal bookkeeping.
-	var fieldsToUpdate []string
-	accountPartial := db.Account{
-		ID: *request.ID,
-	}
-	if request.AdminRoleArn != nil {
-		fieldsToUpdate = append(fieldsToUpdate, "AdminRoleArn")
-		accountPartial.AdminRoleArn = *request.AdminRoleArn
-	}
-	if request.Metadata != nil {
-		fieldsToUpdate = append(fieldsToUpdate, "Metadata")
-		accountPartial.Metadata = *request.Metadata
-	}
-	if len(fieldsToUpdate) == 0 {
-		response.WriteRequestValidationError(
-			w,
-			fmt.Sprintf("Unable to update account %s: "+
-				"no updatable fields provided",
-				accountID),
-		)
+	var amSvc accountmanageriface.AccountManagerAPI
+	if err := Services.Config.GetService(&amSvc); err != nil {
+		api.WriteAPIErrorResponse(w, err)
 		return
 	}
 
-	// Update the DB record
-	acct, err := Dao.UpdateAccount(accountPartial, fieldsToUpdate)
+	a, err := account.GetAccountByID(accountID, dataSvc, dataSvc)
 	if err != nil {
-		// If the account doesn't exist, return a 404
-		if _, ok := err.(*db.NotFoundError); ok {
-			response.WriteNotFoundError(w)
-			return
-		}
-		// Other DB errors return a 500
-		log.Printf("ERROR: Failed to update account %s: %s", *request.ID, err)
-		response.WriteServerErrorWithResponse(w, "Internal Server Error")
+		api.WriteAPIErrorResponse(w, err)
 		return
 	}
-
-	accountJSON, err := json.Marshal(response.AccountResponse(*acct))
+	err = a.Update(d, amSvc)
```

Comment:
Does `Update()` mutate the account object? If that's the case, it might just be worth a comment in the method docs.

Reply:
It changes the data and saves it.  I've been debating if that should be two unique steps.